test-runner/src/main/scala/com/typesafe/spark/test/mesos/RolesSpec.scala

```diff
@@ -1,61 +1,71 @@
 package com.typesafe.spark.test.mesos
 
+import org.apache.spark.SparkContext
 import org.scalatest.Assertions._
 
 import com.typesafe.spark.test.mesos.mesosstate.MesosCluster
 import com.typesafe.spark.test.mesos.framework.runners.RoleConfigInfo
 
 
-trait RolesSpec { self: MesosIntTestHelper =>
+trait RolesSpec extends RoleSpecHelper {
+  self: MesosIntTestHelper =>
 
   def mesosConsoleUrl: String
 
   def cfg: RoleConfigInfo
 
-  runSparkTest("simple count in fine-grain mode with role",
-    "spark.mesos.coarse" -> "false", "spark.mesos.role" -> cfg.role, "spark.cores.max" -> cfg.roleCpus) { sc =>
-    val rdd = sc.makeRDD(1 to 5)
-    val res = rdd.sum()
+  runSparkTest("simple count in fine-grained mode with role",
+    "spark.mesos.coarse" -> "false", "spark.mesos.role" -> cfg.role) { sc =>
+    testRole(sc)
+  }
 
-    assert(15 == res)
+  runSparkTest("simple count in coarse-grained mode with role",
+    "spark.mesos.coarse" -> "true", "spark.mesos.role" -> cfg.role) { sc =>
+    testRole(sc)
+  }
+}
 
-    val m = MesosCluster.loadStates(mesosConsoleUrl)
-    assert(m.sparkFramework.isDefined, "The driver should be running")
+trait RoleSpecHelper {
+  self: MesosIntTestHelper with RolesSpec =>
 
-    if (cfg.role != "*") {
-      // TODO: add message
-      assert(m.slaves.flatMap { x => x.roleResources.map { y => y.roleName } }.contains(cfg.role))
+  def testRole(sc: SparkContext) : Unit = {
 
-      // TODO: add message
-      assert(m.sparkFramework.get.resources.cpu == cfg.roleCpus.toInt)
+    {
+      val m = MesosCluster.loadStates(mesosConsoleUrl)
+      // pre-conditions
+      if (cfg.role != "*") {
+        assert(m.slaves.flatMap { x => x.roleResources.map { y => y.roleName } }.contains(cfg.role), "Spark role should exist.")
+      }
     }
-  }
 
-  ignoreSparkTest("simple count in coarse-grained mode with role",
-    "spark.mesos.coarse" -> "true", "spark.mesos.role" -> cfg.role, "spark.cores.max" -> cfg.roleCpus) { sc =>
-    val rdd = sc.makeRDD(1 to 5)
-    val res = rdd.sum()
+    val expectedUsedCpus = {
+      val m = MesosCluster.loadStates(mesosConsoleUrl)
+      val tmp = m.slaves.map { x => x.resources.cpu }.sum
+      if (this.isInstanceOf[ClusterModeSpec]) {tmp - 1} else tmp  // minus the cpus owned by the Spark Cluster
+    }
 
-    assert(15 == res)
+    val mesosUrl = mesosConsoleUrl
+    val accum = sc.accumulator[Double](0L, "cpu_counter")
 
-    val m = MesosCluster.loadStates(mesosConsoleUrl)
-    assert(m.sparkFramework.isDefined, "The driver should be running")
+    // use enough tasks to utilize all expected resources for tested role and *, pick a task in the middle and observe resources
+    val rdd = sc.makeRDD(1 to 100, 20).mapPartitionsWithIndex{ (p, i) =>
 
-    if (cfg.role != "*") {
-      // TODO: add message
-      assert(m.slaves.flatMap { x => x.roleResources.map { y => y.roleName } }.contains(cfg.role))
+      var counter = 0
```

Comment:
Is this counter a way to break the loop in case it takes too long to acquire all executors?